src/core/server/integration_tests/config/config_deprecation.test.ts

```diff
@@ -24,13 +25,19 @@ describe('configuration deprecations', () => {
     }
   });
 
-  it('should not log deprecation warnings for default configuration', async () => {
-    root = createRoot();
+  if (getFips() === 0) {
+    it('should not log deprecation warnings for default configuration', async () => {
+      root = createRoot();
 
-    await root.preboot();
-    await root.setup();
+      await root.preboot();
+      await root.setup();
 
-    const logs = loggingSystemMock.collect(mockLoggingSystem);
-    expect(logs.warn.flat()).toHaveLength(0);
-  });
+      const logs = loggingSystemMock.collect(mockLoggingSystem);
+      expect(logs.warn.flat()).toHaveLength(0);
+    });
+  } else {
+    it('fips is enabled and the default configuration has been overridden', () => {
+      expect(getFips()).toBe(1);
+    });
+  }
```

Comment:
Was this test logging warnings? Also, FMI, is the intention that in future fips enabled will be the default?

Reply:
> Was this test logging warnings?

It fails since I have `oss=false` because I need the Security Plugin to be registered to allow `xpack.security.experimental.fipsMode.enabled`, otherwise `ensureValidConfiguration` check in preboot/setup throw an error.

<img width="2483" alt="Screenshot 2024-11-12 at 9 58 36 PM" src="https://github.com/user-attachments/assets/66a5a26e-13c9-4a19-a4d0-2e0e03eed2a4">

> Also, FMI, is the intention that in future fips enabled will be the default?

I do not have an answer at this time. FIPS requires a trial/platinum license at this point, which might not work in our current basic tier offering

> is the intention to wind back these changes when FIPs is more broadly enabled?

It Depends. Since in FIPS mode, requires a trial license and (for now) the security plugin having been registered, we would have to find a better way to provide these overrides, rather than just disabling oss. When we move the security config to core security, I could see revisiting these tests.

The goal for the tests running in the FIPS pipeline is not to check if the functionality works as intended but to maximize the code run with Node using a FIPS compliant OpenSSL.